megatron/megatron/training/initialize.py

```diff
@@ -280,6 +280,11 @@ def _initialize_distributed(get_embedding_ranks, get_position_embedding_ranks):
             'rank': args.rank,
             'timeout': timedelta(minutes=args.distributed_timeout_minutes),
         }
+        
+        if args.enable_hetero:
+            # if not all(device_type == args.hetero_device_types[0] for device_type in args.hetero_device_types):
+            #     init_process_group_kwargs['backend'] = 'gloo'
+            init_process_group_kwargs['backend'] = 'gloo'
```

Comment:
This can be confusing because we use both `gloo` and `nccl`, but here only `gloo` is used as the backend.

Reply:
The default global group can not be initialized if there are different devices.